### PR TITLE
add touch events handling to slider

### PR DESCRIPTION
**Pull request by martynovs**
#### PR checklist

- [x] New feature, bugfix, or enhancement
  - [x] Includes tests

#### What changes did you make?

I've added touch events handling to slider component, so it can be used for mobile browsers.

#### Is there anything you'd like reviewers to focus on?

Touch events are fired instead of mouse events in mobile browsers,
to simulate this behavior on desktop you can use device mode in Chrome
https://developers.google.com/web/tools/chrome-devtools/device-mode/

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @martynovs! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by adidahiya** [comment]
Nice, this looks pretty sweet @martynovs! Any chance you could add unit tests here? I'm also not sure why Circle isn't building your PR (it should automatically build forks under our CircleCI project) -- can you try adding a new commit that merges origin/master?

**Comment by martynovs** [comment]
@adidahiya I've added tests.
Some reason test runner take long time to close process after all tests are passed.
It take 18s to perform all core tests on my notebook, but process ended after 49s, maybe it's related to PhantomJS.

**Comment by martynovs** [comment]
@adidahiya I've added tests.

circleci failed to build project because of unrelated error in table/index.ts

File: packages/core/src/components/slider/slider.tsx [hunk 1]
```diff
@@ -72,12 +72,18 @@ export class Slider extends CoreSlider<ISliderProps> {
         );
     }
 
-    protected handleTrackClick(event: React.MouseEvent<HTMLElement>) {
+    protected handleTrackClick(event: MouseEvent | React.MouseEvent<HTMLElement>) {
```
Comment (adidahiya) [inline packages/core/src/components/slider/slider.tsx]: why do we need to allow regular mouse events in addition to react's synthetic mouse events here (and in a few other places)?

Comment (martynovs) [inline packages/core/src/components/slider/slider.tsx]: This signature is declared in CoreSlider

Comment (martynovs) [inline packages/core/src/components/slider/slider.tsx]: I've checked code, handleTrackClick/Touch is protected and will newer be called with native events, it's safe to remove it.

**Comment by llorca** [comment]
Great stuff @martynovs 👏

**Pull request merged**